tls-client-pelion/mbedtls_mbedos_pelion_config.h

```diff
@@ -0,0 +1,283 @@
+//----------------------------------------------------------------------------
+// The confidential and proprietary information contained in this file may
+// only be used by a person authorised under and to the extent permitted
+// by a subsisting licensing agreement from ARM Limited or its affiliates.
+//
+// (C) COPYRIGHT 2016 ARM Limited or its affiliates.
+// ALL RIGHTS RESERVED
+//
+// This entire notice must be reproduced on all copies of this file
+// and copies of this file may only be made by a person if such person is
+// permitted to do so under the terms of a subsisting license agreement
+// from ARM Limited or its affiliates.
+//----------------------------------------------------------------------------
+
+#ifndef PAL_MBEDTLS_USER_CONFIG_H
+#define PAL_MBEDTLS_USER_CONFIG_H
+
+
+/*! All of the following definitions are mandatory requirements for correct 
+*   functionality of PAL TLS and Crypto components.
+*   Please do not disable them.
+*/
+
+/* Platform has time function to provide time for certificates verifications */
+#if 1 //Please set to 1 if you are using secure time
+#ifndef MBEDTLS_HAVE_TIME
+    #define MBEDTLS_HAVE_TIME
+#endif //MBEDTLS_HAVE_TIME
+
+#ifndef MBEDTLS_HAVE_TIME_DATE
+    #define MBEDTLS_HAVE_TIME_DATE
+#endif //MBEDTLS_HAVE_TIME_DATE
```

Comment:
Are we using secure time? Does this work with the library/software you are using? If I recall correctly, there were some issues with ARMCC because Mbed OS did not provide an implementation for gmtime() and ARMCC does not support it.